modules/swagger-codegen/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -86,12 +79,8 @@ export class {{classname}} {
      {{/summary}}
      {{#allParams}}* @param {{paramName}} {{description}}
      {{/allParams}}*/
-{{^isResponseFile}}
-    public {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}extraHttpRequestParams?: any): Observable<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}{}{{/returnType}}> {
-{{/isResponseFile}}
-{{#isResponseFile}}
-    public {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}extraHttpRequestParams?: any): Observable<{{#returnType}}{{{returnType}}}|undefined{{/returnType}}{{^returnType}}{}{{/returnType}}> {
-{{/isResponseFile}}
+    {{! if you change this method signature, also change the version below }}
+    public {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}{{^useHttpClient}}extraHttpRequestParams?: RequestOptionsArgs{{/useHttpClient}}): Observable<{{#returnType}}{{{returnType}}}{{#isResponseTypeFile}}|undefined{{/isResponseTypeFile}}{{/returnType}}{{^returnType}}{}{{/returnType}}> {
```

Comment:
This causes to always have a trailing comma in the end of all of the parameters.

Reply:
As mentioned above it produces this result:
```
getBillingInformation(start: Date, end: Date,): Observable<Array<BillingInformation>>{}
downloadLogoForCompany(companyId: string, fetchThumb: boolean,): Observable<ArrayBuffer>{}
```

Reply:
Yup that definitely needs to be fixed then.

Reply:
@krizic thanks for testing the change.

I think the following should fix the issue
```
    public {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}{{#hasMore}}, {{/hasMore}}{{/allParams}}{{^useHttpClient}}, extraHttpRequestParams?: RequestOptionsArgs{{/useHttpClient}}): Observable<{{#returnType}}{{{returnType}}}{{#isResponseTypeFile}}|undefined{{/isResponseTypeFile}}{{/returnType}}{{^returnType}}{}{{/returnType}}> {
 ```

When you've time, I wonder if you can give it a try by replacing line 83 with it.

cc @bedag-moo

Reply:
I can confirm this is still an issue, and if you see the code it is obvious that in case that `useHtpClient` flag is true he WON'T add last method parameter (extraHttpRequestParams) effectively leaving method signature to always end with `,`.

`{{^useHttpClient}}*extraHttpRequestParams?:RequestOptionsArgs{{/useHttpClient}}`

Reply:
@wing328 thanks for providing a quick solution!

Reply:
I added `{{#hasMore}}, {{/hasMore}}` to skip "," for the last parameter. Are you using `-t` parameter to supply the customized templates with the fix?

Reply:
Otherwise, you will need to do a "mvn clean package" to recompile the JAR.

Reply:
@wing328 you are right. I wasn't cleaning bins. Works now, thanks for the solution again. I hope this gets merged soon

Reply:
@krizic np. Thanks for confirming the fix works.